### PR TITLE
Android: Minor Code Improvements

### PR DESCRIPTION
**Pull request by oong819**
Add compact, short information about your PR for easier understanding:

### Goal of the PR
improve Android build code quality
### How does the PR work?
various code improvements
### Does it resolve any reported issue?
not yet checked, but seems not
### Does this relate to a goal in [the roadmap](../doc/direction.md)?
no
### If not a bug fix, why is this PR needed? What usecases does it solve?
remove dead code which probally Proguard already do the job, add needed permission, etc.

## To do

Ready for Review.

## How to test
1. install on an Android device (SDK 21+)
2. clear Minetest's app data (BACKUP FIRST)
3. everything should be fine and the Minetest app should be fullscreen and a notification should appear when Minetest unzipping stuff

### CODE REVIEW

**Comment by oong819** [comment]
> 

oooh, i will try my best to resolve those

**Comment by oong819** [comment]
also is AppCompat really needed?

**Comment by srifqi** [comment]
AndroidX (was Android Support Library) is essential in ensuring compatibility between Android versions.
- Blog post: https://android-developers.googleblog.com/2018/05/hello-world-androidx.html
- AndroidX page: https://developer.android.com/jetpack/androidx?hl=en

File: android/app/src/main/java/net/minetest/minetest/GameActivity.java [hunk 1]
```diff
@@ -64,7 +63,6 @@ public void onCreate(Bundle savedInstanceState) {
 	}
 
 	private void makeFullScreen() {
-		if (Build.VERSION.SDK_INT >= 19)
 			this.getWindow().getDecorView().setSystemUiVisibility(
 					View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION |
 					View.SYSTEM_UI_FLAG_HIDE_NAVIGATION |
```
[the remark this answers is not in the document]
Comment (oong819) [reply android/app/src/main/java/net/minetest/minetest/GameActivity.java]: is it okay now?

**Comment by oong819** [comment]
i need help resolve a/android typo, idk where the typo at

**Comment by srifqi** [comment]
I can not find the typo that you mention other than people's names. Note that I am limiting the scope to be just the Android portion of the code.

**Review by srifqi — COMMENTED** [review]
This PR already covers problems that I can find with helps from Android Studio.

**Comment by oong819** [comment]
wait i fixed  --- a/android/app/src/main/AndroidManifest.xml
+++ b/android/app/src/main/AndroidManifest.xml

**Comment by oong819** [comment]
aaah miss tap

**Comment by oong819** [comment]
if everything are ok, approve pls?

**Review by srifqi — COMMENTED** [review]
This PR already covers problems that I can find with helps from Android Studio. (no change from before)

**Comment by oong819** [comment]
lets wait...
now with test video: https://youtu.be/7Gmz73oqfCg

**Review by rubenwardy — APPROVED** [review]

**Comment by Ronny2508** [comment]
> Add compact, short information about your PR for easier understanding:
> 
> ### Goal of the PR
> improve Android build code quality
> 
> ### How does the PR work?
> various code improvements
> 
> ### Does it resolve any reported issue?
> not yet checked, but seems not
> 
> ### Does this relate to a goal in [the roadmap](../doc/direction.md)?
> no
> 
> ### If not a bug fix, why is this PR needed? What usecases does it solve?
> remove dead code which probally Proguard already do the job, add needed permission, etc.
> 
> ## To do
> Ready for Review.
> 
> ## How to test
> 1. install on an Android device (SDK 21+)
> 2. clear Minetest's app data (BACKUP FIRST)
> 3. everything should be fine and the Minetest app should be fullscreen and a notification should appear when Minetest unzipping stuff

**Pull request merged**